test/GroupByTest.cpp

```diff
@@ -1521,30 +1521,16 @@ TEST_F(GroupByOptimizations, computeGroupByForJoinWithFullScan) {
     // Set up a GROUP BY operation for which the optimization can be applied.
     // The last two arguments of the `Join` constructor are the indices of the
     // join columns.
-    IdTable result(qec->getAllocator());
     auto join = makeExecutionTree<Join>(qec, values, xyzScanSortedByX, 0, 0);
     GroupBy validForOptimization{qec, variablesOnlyX, aliasesCountX, join};
-    if (chooseInterface) {
-      ASSERT_TRUE(
-          validForOptimization.computeGroupByForJoinWithFullScan(&result));
-    } else {
-      ASSERT_TRUE(
-          validForOptimization.computeOptimizedGroupByIfPossible(&result));
-    }
-
-    // There are 7 triples with `<x>` as a subject, 0 triples with `<xa>` as a
-    // subject, and 1 triple with `y` as a subject.
-    ASSERT_EQ(result.numColumns(), 2u);
-    ASSERT_EQ(result.size(), 2u);
-
-    auto getId = makeGetId(qec->getIndex());
-    Id idOfX = getId("<x>");
-    Id idOfY = getId("<y>");
-
-    ASSERT_EQ(result(0, 0), idOfX);
-    ASSERT_EQ(result(0, 1), Id::makeFromInt(7));
-    ASSERT_EQ(result(1, 0), idOfY);
-    ASSERT_EQ(result(1, 1), Id::makeFromInt(1));
+    auto optional =
+        chooseInterface
+            ? validForOptimization.computeGroupByForJoinWithFullScan()
+            : validForOptimization.computeOptimizedGroupByIfPossible();
+    EXPECT_THAT(
+        optional,
+        Optional(Eq(W{makeIdTableFromVector(
+            {{idOfX, Id::makeFromInt(7)}, {idOfY, Id::makeFromInt(1)}})})));
```

Comment:
I think it would be easy + beneficial, also for the rest of the codebase,
To implement in the `IdTableHelpers` file a `static` lambda (or a callable, or maybe a function, you choose),
1. that is called `matchIdTable`, takes an IdTable, and returns `Eq(W(std::move(table))`
2. that is called `matchIdTableFromVector` that takes exactly the arguments of `makeIdTableFromVector`.

Reply:
the first function should of course take variadic arguments from which an `IdTable` can be constructed.